src/modules/0.0.7/aws_route53_hosted_zones/index.ts

```diff
@@ -18,8 +18,8 @@ export const AwsRoute53HostedZoneModule: Module2 = new Module2({
     },
     resourceRecordSetMapper: async (rrs: any, ctx: Context) => {
       const out = new ResourceRecordSet();
-      // if (!(rrs.Name && rrs.Type)) throw new Error('Wrong record from AWS');
-      if (!(rrs.Name && rrs.Type)) return undefined;
+      if (!(rrs.Name && rrs.Type)) throw new Error('Wrong record from AWS');
+      // if (!(rrs.Name && rrs.Type)) return undefined;
```

Comment:
@aguillenv since you wrote this module originally. Do you have any idea why *ignoring* this busted data from AWS would actually break things?

Reply:
Those properties must always be present. This is just an extra check in case AWS glitch or something but those properties should always be present. The name because it must have at least the base domain name and all records have a type `A|AAAA|TXT|NS|...`

Reply:
Right, but I mean, if the record we get back from AWS is broken, my change (before commenting it out and reverting back) was just dropping the record entirely from consideration, and that was causing an *update* of the domain name to fail.

Allowing the bad data to come through and throw an error is fixing things, but I don't see why just ignoring the bad data isn't good enough here?

Reply:
ok if I'm not wrong what is happening is the following:
- The replace is triggered. The hosted zone replace just creates the new one and let the next loop delete the previous on
- In the same first iteration is trying to create the new records and delete the old ones
- In the next loop it is trying to delete the old hosted zone, but it doing something weird trying to modify/update the records, but some of these ones belonged to the old one hosted zone and they do not exists anymore, so it fails in the update 

My guess is that it is passing with the old version because the `throw` forces a new loop where all hosted zones are already in place?

I think I need to review this logic. I'll do it since I still need to update this module for the elb+r53 integration